### PR TITLE
Update world map

### PR DESCRIPTION
**Pull request by AliceR**
This change updates the map component.

Still missing features:

- [x] click and popover on map points
- [x] use mvt instead of `/locations` to display on the map
- [x] update map on filter select (dropdown in legend) #488 
- [x] reuse map component for "nearby locations" on location page
- [-] filter mvt for nearby locations
- [separate issue] scale up approach to layer styles to more parameters (currently a fixed definition in [utils/map-settings.js](https://github.com/openaq/openaq.org/blob/develop/app/assets/scripts/utils/map-settings.js))

### CODE REVIEW

File: app/assets/scripts/components/map/legend.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  let drop = (
+    <Dropdown
+      triggerElement="button"
+      triggerClassName="button button--primary-unbounded drop__toggle--caret"
+      triggerTitle="Show/hide parameter options"
+      triggerText={activeParameter.name}
+    >
+      <ul role="menu" className="drop__menu drop__menu--select">
+        {parameters.map((o, i) => (
```
[the remark this answers is not in the document]
Comment (AliceR) [reply app/assets/scripts/components/map/legend.js:22]: good catch! I need to clean up the parameter situation :/

[the remark this answers is not in the document]
Comment (AliceR) [reply app/assets/scripts/components/map/legend.js:23]: this should be fixed - but selecting one of the options won't do anything yet. Another task on the to do list...

File: app/assets/scripts/components/map/popover.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                href={`#/compare/${encodeURIComponent(id)}`}
+                className="button button--primary-bounded"
+                title={`Compare ${name} with other locations`}
+              >
+                Compare
+              </a>
+            </li>
+            <li>
+              <a
+                href={`#/location/${encodeURIComponent(id)}`}
```
[the remark this answers is not in the document]
Comment (AliceR) [reply app/assets/scripts/components/map/popover.js]: should be fixed

**Review by jvntf — CHANGES_REQUESTED** [review]
Last thing, it seems that on click of 'view more' it seems that the url updates but the page does not load. i.e. url is populated with a new location id but the previous location page remains rendered until the browser is refreshed. My guess is this is related to preexisting issues with the Router?

**Comment by AliceR** [comment]
huh weird, it works for me... can you check for errors in the console, please?

**Comment by jvntf** [comment]
Ah sorry i was looking at the NearbyLocations map. its the same component right? @AliceR

**Comment by jvntf** [comment]
if so, im seeing several errors on single location page. if not, apologies for mixing up discussion!

**Comment by necoline** [comment]
I'm seeing the same errors on the location page 
```Error {message: "http status 200 returned without content."}```
is this what you are seeing @jvntf ?

**Comment by jvntf** [comment]
earlier i thought i had seen other errors but now i just see that no content error, and some propTypes errors

**Comment by AliceR** [comment]
you are right, the map on the location page has some issues. investigating...

**Comment by AliceR** [comment]
@necoline , @jvntf , could you test again, please? Those `Error {message: "http status 200 returned without content."}` can be ignored, afaik thats just some artifact for empty tiles.
The source/layer are not updating reliably, I think. Sometimes it's just loading slowly, sometimes the api fails with 500... hard to say if thats a problem with the code or the api itself.

**Comment by AliceR** [comment]
A loading animation would be nice, but I didn't find a straightforward solution to add based on map events https://docs.mapbox.com/mapbox-gl-js/api/map/#map-events

File: app/assets/scripts/components/map/legend.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import { useHistory } from 'react-router-dom';
+import { Dropdown } from 'openaq-design-system';
+import c from 'classnames';
+
+import { generateLegendStops } from '../../utils/colors';
+
+export default function Legend({ parameters, activeParameter }) {
+  let history = useHistory();
+
+  function onFilterSelect(parameter, e) {
```
Comment (jvntf) [inline app/assets/scripts/components/map/legend.js:12]: image
seeing this error from world map

**Review by jvntf — CHANGES_REQUESTED** [review]
Now just seeing this one error on world map page. single location looks good other than the empty tiles and some propTypes errors

File: app/assets/scripts/components/map/legend.js (hunk 3 above)
Comment (AliceR) [reply app/assets/scripts/components/map/legend.js:13]: wonder how history can be undefined, thats coming from react router 🤔 I tried, but never saw this error myself

Comment (jvntf) [reply app/assets/scripts/components/map/legend.js:13]: must be a rebuild issue or something then

**Review by necoline — APPROVED** [review]

**Pull request merged**